src/common/parse-util.c

```diff
@@ -0,0 +1,43 @@
+#include "parse-util.h"
+#include <errno.h>
+#include <inttypes.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+bool parse_long(const char *in, long *ret) {
+        const int base = 10;
+        char *x = NULL;
```

Comment:
Not sure if `diff` is better, but `x` is a bit too generic. 
```suggestion
        char *diff = NULL;
```

Reply:
I took inspiration from [systemd/src/basic/parse-util.c](https://github.com/systemd/systemd/blob/v238/src/basic/parse-util.c). This form of variable naming is very common in many C projects, not only in systemd, when it comes to small general functions like this one. It creates neat, readable code. 
Of course in more complicated code the variables should have more meaningful names

Reply:
Its also quite common in Go projects - and quite often causing code that is actually harder to read. In our case the  condition 
```C
if (errno > 0 || !x || x == in || *x != 0) {
```
requires more mental overhead. 
But as you said, the scope of the variable is pretty small and, therefore, its ok.

Reply:
That was my point. Only in small and simple functions.
As for the condition - yes, but on the other hand it keeps the condition in one line, where if the variable was long and descriptive - it would take multiple lines. Here it is not necessary.